src/cache_cf.cc

```diff
@@ -3072,7 +3072,7 @@ free_time_msec(time_msec_t * var)
 static void
 dump_time_nanoseconds(StoreEntry *entry, const char *name, const std::chrono::nanoseconds &var)
 {
-    storeAppendPrintf(entry, "%s %" PRId64 " nanoseconds\n", name, var.count());
+    storeAppendPrintf(entry, "%s %" PRId64 " nanoseconds\n", name, static_cast<int64_t>(var.count()));
```

Comment:
What type is FreeBSD using for this value?

FYI, it already smells that a counter is using a signed type. Most likely this should be displaying PRIu64 or similar.

Please at least add a comment indicating this cast is a hack and if possible for which OS version. So we can purge it in future when no longer needed.

Reply:
```
cache_cf.cc:3075:68: error: format specifies type 'long' but the argument has type 'std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000000> >::rep' (aka 'long long') [-Werror,-Wformat]
    storeAppendPrintf(entry, "%s %" PRId64 " nanoseconds\n", name, var.count());

```
interestingly enough, both long and long long are 64-bit, but clang on freebsd automatically convert.
switching to PRIu64 doesn't help:
```
depbase=`echo cache_cf.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`; ccache clang++90 -DHAVE_CONFIG_H -DDEFAULT_CONFIG_FILE=\"/home/jenkins/squid/install/etc/squid.conf\"  -DDEFAULT_SQUID_DATA_DIR=\"/home/jenkins/squid/install/share\"  -DDEFAULT_SQUID_CONFIG_DIR=\"/home/jenkins/squid/install/etc\"    -I.. -I../include  -I../lib -I../src  -I../include -I/usr/local/include   -I../libltdl -I../src -I../libltdl   -I/usr/local/include/libxml2 -Werror -Qunused-arguments  -D_REENTRANT -I/usr/local/include/libxml2 -g -O2 -I/usr/local/include -MT cache_cf.o -MD -MP -MF $depbase.Tpo -c -o cache_cf.o cache_cf.cc && mv -f $depbase.Tpo $depbase.Po
cache_cf.cc:3075:68: error: format specifies type 'unsigned long' but the argument has type 'std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000000> >::rep' (aka 'long long') [-Werror,-Wformat]
    storeAppendPrintf(entry, "%s %" PRIu64 " nanoseconds\n", name, var.count());
                                 ~~~~~~~~~                         ^~~~~~~~~~~
1 error generated.
```